src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -71,7 +72,15 @@ public static void Fill<T>(ref T refData, nuint numElements, T value)
                 }
                 else if (Unsafe.SizeOf<T>() == 32)
                 {
-                    vector = Unsafe.As<T, Vector256<byte>>(ref tmp).AsVector();
+                    if (Vector<byte>.Count == 32)
+                    {
+                        vector = Unsafe.As<T, Vector256<byte>>(ref tmp).AsVector();
+                    }
+                    else
+                    {
+                        Debug.Fail("Vector<T> isn't 256 bits in size?");
```

Comment:
This shouldn't be a failure (unless I'm missing some other condition somewhere). It is valid to have a `Vector256<T>` on say ARM, where `Vector<T>` is only ever `16-bytes`.

It's just that the given `Vector256<T>` will have no hardware acceleration and will be entirely software driven.

Reply:
Likewise on a machine with AVX support but no AVX2 support (partial hardware acceleration for `V256`) or a machine without any AVX support (no hardware acceleration for `V256`)

Reply:
There's already a condition at the top of the method which skips the vectorized code path if `Unsafe.SizeOf<T>() > Vector<byte>.Count`. This check is intended to catch `Vector<T>` unexpectedly growing to - say - 512 bits.

Reply:
Thanks, that would be the "I'm missing some other condition somewhere" case 😄